config/pmd-test.xml

```diff
@@ -191,4 +191,15 @@
         </properties>
     </rule>
 
+    <rule ref="category/java/design.xml/UseObjectForClearerAPI">
+        <properties>
+            <!-- Suppress the warning of rule UseObjectForClearerAPI which would report
+                    method createSuppressionXpathSingleFilter contains too much String
+                    param in SuppressionXpathSingleFilter. -->
```

Comment:
You pretty much restate what the xpath suppression says. We want to know reason for suppression, besides "there was a violation".

Lets have something like we prefer to be specific in what options are required in test area and have them written out.

Reply:
Sorry, I cannot quite understand what's the meaning of it. Aren't these lines of comments just for explaining what kinds of CI error it would suppress (which the methods contain too much String params)?

Reply:
> these lines of comments just for explaining what kinds of CI error it would suppress 

We already know from the category and xpath the type of violation we are suppressing. We want an overview of why it is being suppressed.
Violation is in `UseObjectForClearerAPI`, how do we get a cleaner API from suppressing violation which is meant to do that.

Take https://github.com/checkstyle/checkstyle/blob/master/config/pmd-test.xml#L25-L27 as an example. Violation is too many methods in class. The reason we suppress this is we prefer to keep all tests related to a single class in a single test so they are all together. It also helps with pitest profiles. By your own example writeup, you would have written something like "they are too many methods in classes which would cause the warning of rule TooManyMethods", but it doesn't give users reading this for the first time the background as to why it is disabled. It just basically restates the check gave a violation and we want it suppressed. It needs to go further so users can understand our reasoning and help them going forward in our project.

Does this make it clearer?

This is why I wrote: "we prefer to be specific in what options are required in test area and have them written out.".
`String...` parameters would cause confusion going forward in the tests. How many parameters should I give? Is it valid to give 2, 5, or none? The method required a specific number of parameters, but `String...` doesn't give user this understanding from just the signature. We would have to expand the javadoc, which most people don't read, or have the user always read the code to understand what the method is truely requiring.

Reply:
Thx for such a detailed explain. I would correct it.